andela/ah-bird-box
# #163388253 feature create user profile

#### What does this PR do?
Have profile for a user upon registration
#### Description of Task to be completed?
- write unit tests
- create models, views, serializers and renderers for profile
- register profiles URL
#### How should this be manually tested?
* clone the repo and `cd` into it.
* checkout to the branch
  `git checkout ft-users-create-profile-163388253`

* `pip install -r requirements.txt`

* Run server
  `python manage.py runserver`

* test on postman using
-  the url `http://127.0.0.1:8000/api/users/` to register an author

- Using the token provided upon registration, add it on authorization header and use the url `http://127.0.0.1:8000/api/profiles/<str: username>` to view an author

- the url `http://127.0.0.1:8000/api/profiles` to view all authors

- the url ` http://127.0.0.1:8000/api/profiles/edit/<str:username>` to edit a profile. Add token on the authorization header and input the following on the form data. 

N.B You cannot edit another author's profile. 

![image](https://user-images.githubusercontent.com/42655718/52702930-4ae69e00-2f8e-11e9-9aba-2828846e8922.png)

* Updated profile should appear

#### Any background context you want to provide?
This must be an authorized user, therefore, a token is added on the authorization header
![image](https://user-images.githubusercontent.com/42655718/52683783-0559af00-2f55-11e9-89a1-7c955e0d0f5f.png)

#### What are the relevant pivotal tracker stories?
[#163388253](https://www.pivotaltracker.com/n/projects/2240063)
#### Screenshots (if appropriate)
 View all authors 
![image](https://user-images.githubusercontent.com/42655718/52635491-0ea63580-2edb-11e9-934e-244afe1822f9.png)
View a specific author
![image](https://user-images.githubusercontent.com/42655718/52665343-6c0ea680-2f1c-11e9-96bc-4da9e7f021f3.png)
Edit your profile
![image](https://user-images.githubusercontent.com/42655718/52702612-90569b80-2f8d-11e9-9f04-5ec1244a8432.png)




#### Questions:
